pyproject.toml

```diff
@@ -1,3 +1,73 @@
 [build-system]
-requires = ["jupyter_packaging>=0.10,<2", "jupyterlab==3.*", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupyter-scatter"
+description = "A scatter plot extension for Jupyter Notebook and Lab"
+authors = [
+  { name = "Fritz Lekschas", author_email = "code@lekschas.de" }
+]
+license = { text = "Apache-2.0" }
+readme = "README.md"
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Multimedia :: Graphics",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+keywords = [
+    "scatter",
+    "scatter plot",
+    "jupyter",
+    "ipython",
+    "ipywidgets",
+    "jupyterlab",
+    "jupyterlab-extension",
+    "widgets",
+]
+dependencies = [
+    "anywidget>=0.2.0",
+    "ipywidgets>=7.6,<9",
+    "ipython",
+    "matplotlib",
+    "numpy",
+    "pandas",
+    "traittypes>=0.2.1",
+    "typing_extensions",
+]
+dynamic = ["version"]
+
+[project.urls]
+homepage = "https://github.com/flekschas/jscatter"
+
+[project.optional-dependencies]
+test = [
+    "jupyter",
+    "jupyterlab>=3.0",
+    "pytest",
+    "flake8",
+]
+
+# Configures hatchling build system to use the verison
+# found in the package.json file for the python package.
+[tool.hatch.version]
+path = "js/package.json"
+pattern = "\"version\": \"(?P<version>.+?)\""
+
+
+[tool.hatch.build.hooks.jupyter-builder]
+build-function = "hatch_jupyter_builder.npm_builder"
+ensured-targets = ["jscatter/bundle.js"]
+skip-if-exists = ["jscatter/bundle.js"]
+dependencies = ["hatch-jupyter-builder>=0.5.0"]
+
+[tool.hatch.build.hooks.jupyter-builder.build-kwargs]
+npm = "npm"
+build_cmd = "build"
+path = "js"
```

Comment:
This is a plugin for hatchling that runs the `npm_builder` when you run `pip install -e .` or `python -m build`. If `jscatter/bundle.js` exists (what we publish to pypi), then the npm build step is skipped.